littlepay/api/funding_sources.py

```diff
@@ -25,6 +28,26 @@ class FundingSourceResponse:
     token_key_id: Optional[str] = None
     icc_hash: Optional[str] = None
 
+    @classmethod
+    def from_kwargs(cls, **kwargs):
+        return from_kwargs(cls, **kwargs)
+
+    def __post_init__(self):
+        """Parses any date parameters into Python datetime objects.
+
+        For @dataclasses with a generated __init__ function, this function is called automatically.
+
+        Includes a workaround for Python 3.10 where datetime.fromisoformat() can only parse the format output
+        by datetime.isoformat(), i.e. without a trailing 'Z' offset character and with UTC offset expressed
+        as +/-HH:mm
+
+        https://docs.python.org/3.11/library/datetime.html#datetime.datetime.fromisoformat
+        """
+        if self.created_date:
+            self.created_date = datetime.fromisoformat(self.created_date.replace("Z", "+00:00", 1))
+        else:
+            self.created_date = None
```

Comment:
Would it work to have `FundingSourceResponse` use the `FundingSourceDateFields` mixin/class instead of re-implementing this logic for the one field?

Reply:
I thought about doing that, but was thinking it might be misleading to imply that `expiry_date` is a possible field on this response 🤔. (since expiration only makes sense in the context of some group)

I don't feel too strongly though so am happy to have it use `FundingSourceDateFields` if you'd prefer that

Reply:
All good, we can come back to this refactor if needed.